File: java/parser/src/main/java/gov/lbl/parser/parser/CommentVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Annotation;
import gov.lbl.parser.domain.Comment;

public class CommentVisitor extends modelicaBaseVisitor<Comment> {
    @Override
    public Comment visitComment(modelicaParser.CommentContext ctx) {
        String_commentVisitor string_commentVisitor = new String_commentVisitor();
        String string_comment = ctx.string_comment() == null ? null : ctx.string_comment().accept(string_commentVisitor);
        AnnotationVisitor annotationVisitor = new AnnotationVisitor();
        Annotation annotation = ctx.annotation() == null ? null : ctx.annotation().accept(annotationVisitor);
        return new Comment(string_comment, annotation);
    }
}

File: jsParser/parser/CommentVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Comment = require('../domain/Comment')

const String_commentVisitor = require('./String_commentVisitor')
const AnnotationVisitor = require('./AnnotationVisitor')

class CommentVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitComment (ctx) {
    let string_comment = ''
    let annotation = null

    if (ctx.annotation()) {
      const annotationVisitor = new AnnotationVisitor.AnnotationVisitor()
      annotation = annotationVisitor.visitAnnotation(ctx.annotation())
    }
    if (ctx.string_comment()) {
      const string_commentVisitor = new String_commentVisitor.String_commentVisitor()
      string_comment = string_commentVisitor.visitString_comment(ctx.string_comment())
    }
    return new Comment.Comment(string_comment, annotation)
  }
};

CommentVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitComment = this.visitComment
exports.CommentVisitor = CommentVisitor
